### ISSUE TITLE
No Kubeflow endpoint after redeploying Kubeflow with same name in GCP

### ISSUE THREAD

**Issue by knave07** [kubeflow/kubeflow#5101]
/kind bug

**What steps did you take and what happened:**
I already had Kubeflow running in my cluster in GKE. Sadly It broke (Kubeflow ns got deleted by someone). So I deleted all the resources, including storage with `kfctl`. It was fine for me as I had a backup of all the build files. 
I applied the config file again and it got deployed without any issues. But then the Kubeflow Endpoint was not working. I saw that there was no Endpoint created in GCP and also there was an issue with envoy-ingress.
<img width="485" alt="Screen Shot 2020-07-02 at 21 58 04" src="https://user-images.githubusercontent.com/37116431/86362076-baf2e280-bcaf-11ea-8498-42ab74b60aa0.png">

I tried to again deploy Kubeflow, but this time with different cluster name. It worked fine and also the endpoint was working. I again deleted all the resources, storage using `kfctl delete` command. I redeploy kubeflow with the same name and got the same error as before.  

<img width="1079" alt="Screen Shot 2020-07-02 at 22 18 47" src="https://user-images.githubusercontent.com/37116431/86364867-b3353d00-bcb3-11ea-83dc-95ef2d537ec0.png">

**What did you expect to happen:**
The Kubeflow endpoint should have been created as normal.

**Environment:**

- Kubeflow version: Build commit: 743746b
- kfctl version: kfctl v1.0.2-0-ga476281
- Kubernetes platform: GKE
- Kubernetes version: 1.14.10-gke.42

**Comment by parthmishra** [comment]
This is related to https://github.com/kubeflow/kubeflow/issues/5048

What you need to do is undelete endpoint name: `gcloud endpoints services undelete <kf_endpoint_name>` and then rerun `kfctl apply ...`

**Comment by jlewi** [comment]
Duplicate of #5048 

Thanks @parthmishra 

@knave07 hopefully that unblocked you.

**Comment by jlewi** [comment]
Duplicate of #5048
